plugins/eddn.py

```diff
@@ -633,6 +631,277 @@ def export_journal_entry(self, cmdr: str, is_beta: bool, entry: Mapping[str, Any
             self.send(cmdr, msg)
             self.parent.children['status']['text'] = ''
 
+    def export_journal_generic(self, cmdr: str, is_beta: bool, entry: Mapping[str, Any]) -> None:
+        """
+        Send an EDDN event on the journal schema.
+
+        :param cmdr: the commander under which this upload is made
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to send
+        """
+        msg = {
+            '$schemaRef': f'https://eddn.edcd.io/schemas/journal/1{"/test" if is_beta else ""}',
+            'message': entry
+        }
+        this.eddn.export_journal_entry(cmdr, entry, msg)
+
+    def entry_augment_system_data(self, entry: Dict[str, Any], system_name: str) -> Union[str, Mapping[str, Any]]:
+        """
+        Augment a journal entry with necessary system data.
+
+        :param entry: The journal entry to be augmented.
+        :param system_name: Name of current star system.
+        :param systemname_field_name: Name of journal key for system name.
+        :return: The augmented version of entry.
+        """
+        # If 'SystemName' or 'System' is there, it's directly from a journal event.
+        # If they're not there *and* 'StarSystem' isn't either, then we add the latter.
+        if entry.get('SystemName') is None and entry.get('System') is None and entry.get('StarSystem') is None:
+            entry['StarSystem'] = system_name
+
+        if entry.get('SystemAddress') is None:
+            if this.systemaddress is None:
+                logger.warning("this.systemaddress is None, can't add SystemAddress")
+                return "this.systemaddress is None, can't add SystemAddress"
+
+            entry['SystemAddress'] = this.systemaddress
+
+        if entry.get('StarPos') is None:
+            if not this.coordinates:
+                logger.warning("this.coordinates is None, can't add StarPos")
+                return "this.coordinates is None, can't add StarPos"
+
+            entry['StarPos'] = list(this.coordinates)
+
+        return entry
+
+    def export_journal_fssdiscoveryscan(
+            self, cmdr: str, system: str, is_beta: bool, entry: Mapping[str, Any]
+    ) -> Optional[str]:
+        """
+        Send an FSSDiscoveryScan to EDDN on the correct schema.
+
+        :param cmdr: the commander under which this upload is made
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to send
+        """
+        #######################################################################
+        # Elisions
```

Comment:
Exclusions? I dont believe Elisions is a word

Reply:
Is so!  As discussed on Discord.

Elision is kind of the noun describing something that has been elided.